src/metabase/api/action.clj

```diff
@@ -139,4 +143,28 @@
         (db/update! existing-model id type-row))))
   (first (action/actions-with-implicit-params nil :id id)))
 
+(api/defendpoint POST "/:id/public_link"
+  "Generate publicly-accessible links for this Action. Returns UUID to be used in public links. (If this
+  Action has already been shared, it will return the existing public link rather than creating a new one.) Public
+  sharing must be enabled."
+  [id]
+  (api/check-superuser)
+  (validation/check-has-application-permission :setting)
```

Comment:
why setting permissions are required here? isn't admins always have this permission?

Reply:
Correct, fixed